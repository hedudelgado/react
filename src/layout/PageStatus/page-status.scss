@import "../../css/main";

.pi-page-status {
  text-align: center;

  .title {
    margin: 12px 0 5px 0;
  }

  &.pi-page-status-warning, &.pi-page-status-error {
    .icon {
      display: block;
      height: 61px;
      background-size: 45px 61px;
      background-position: center;
      background-repeat: no-repeat;

      &:before {
        font-family: 'wb-icons';
        font-size: 4rem;
        content: '\e95f';
      }
    }
  }

  &.pi-page-status-warning {
    .icon {
      color: $pi-notification-warning-dark;
    }
  }

  &.pi-page-status-error {
    .icon {
      color: $pi-notification-error-dark;
    }
  }
}
