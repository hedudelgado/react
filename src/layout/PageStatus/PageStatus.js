import React from 'react';
import {
  node,
  oneOf,
  string,
} from 'prop-types';
import classNames from 'classnames';
import BaseStyle from '../BaseStyle';
import Typography from '../Typography';
import Icon from '../Icon';
import './page-status.scss';

const PageStatus = ({
  ariaLive,
  variant,
  title,
  children,
  className,
  ...otherProps
}) => {
  const classes = classNames(
    'pi-page-status',
    `pi-page-status-${variant}`,
    className,
  );

  return title && children && (
    <BaseStyle data-test={`error-${variant}`} role="region" aria-live={ariaLive} component="div" className={classes} {...otherProps}>
      <Icon icon="disc-info-new" className="icon" />
      <Typography component="h5" className="title">
        {title}
      </Typography>
      <div className="message">
        {children}
      </div>
    </BaseStyle>
  );
};

PageStatus.propTypes = {
  /**
   * Variant defines the Page Status type.
   */
  variant: oneOf(['warning', 'error']).isRequired,
  /**
   * Aria-Live is used to set the priority with which a screen reader should treat
   * updates to live regions.
   * The possible settings are: off, polite or assertive.
   */
  ariaLive: oneOf(['polite', 'assertive', 'off']).isRequired,
  /**
   * An optional title in the Page Status.
   */
  title: string.isRequired,
  /**
   * The Page Status message. It is usually a text, but it can also be further DOM.
   */
  children: node,
};

PageStatus.defaultProps = {
  children: null,
};

export default PageStatus;
