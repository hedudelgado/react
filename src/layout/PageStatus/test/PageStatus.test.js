import React from 'react';
import { shallow } from 'enzyme';
import PageStatus from '../PageStatus';

describe('PageStatus component', () => {
  it('should render the warning variant component', () => {
    const props = {
      variant: 'warning',
      ariaLive: 'off',
      title: 'This is a title',
    };
    const wrapper = shallow(<PageStatus {...props}>This is a notification</PageStatus>);

    expect(wrapper).toMatchSnapshot();
  });

  it('should render the error variant component', () => {
    const props = {
      variant: 'error',
      ariaLive: 'off',
      title: 'This is a title',
    };
    const wrapper = shallow(<PageStatus {...props}>This is a notification</PageStatus>);

    expect(wrapper).toMatchSnapshot();
  });
});
