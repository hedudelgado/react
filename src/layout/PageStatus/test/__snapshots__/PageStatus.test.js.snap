// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageStatus component should render the error variant component 1`] = `
<BaseStyle
  aria-live="off"
  className="pi-page-status pi-page-status-error"
  component="div"
  data-test="error-error"
  role="region"
>
  <Icon
    className="icon"
    icon="disc-info-new"
  />
  <Typography
    className="title"
    component="h5"
  >
    This is a title
  </Typography>
  <div
    className="message"
  >
    This is a notification
  </div>
</BaseStyle>
`;

exports[`PageStatus component should render the warning variant component 1`] = `
<BaseStyle
  aria-live="off"
  className="pi-page-status pi-page-status-warning"
  component="div"
  data-test="error-warning"
  role="region"
>
  <Icon
    className="icon"
    icon="disc-info-new"
  />
  <Typography
    className="title"
    component="h5"
  >
    This is a title
  </Typography>
  <div
    className="message"
  >
    This is a notification
  </div>
</BaseStyle>
`;
